app/assets/javascripts/Components/grades_summary_display.jsx

```diff
@@ -35,73 +35,37 @@ class GradesSummaryDisplay extends React.Component {
     this.state = {
       columns: [],
       data: [],
-      loading: true,
-      datasets: [],
       labels: [],
-      xTitle: I18n.t('activerecord.models.assessment.one'),
-      yTitle: I18n.t('activerecord.models.mark.one') + ' (%)'
+      average_data: [],
+      median_data: [],
+      individual_data: [],
+      loading: true
     }
-    this.fetchData = this.fetchData.bind(this);
   }
 
   componentDidMount() {
     this.fetchData();
   }
 
-  fetchData() {
+  fetchData = () => {
     $.ajax({
       url: Routes.populate_course_summaries_path(),
       dataType: 'json',
-    }).then(res => {
-      let labels = Object.keys(res.columns).map(k => {
-        return res.columns[k].Header;
-      });
-      this.averageDataSet.data = [];
-      this.medianDataSet.data = [];
-      labels.forEach(l => {
-        if (res.assessment_info[l]) {
-          this.averageDataSet.data.push(
-            res.assessment_info[l]['average'] ? parseFloat(res.assessment_info[l]['average']) : null
-          );
-          this.medianDataSet.data.push(
-            res.assessment_info[l]['median'] ? parseFloat(res.assessment_info[l]['median']) : null
-          );
-        }
-      });
-      let chartInfo = {labels: labels};
-      if (this.props.student) {
-        this.individualDataSet.data = [];
-        Object.keys(res.columns).forEach(k => {
-          if (res.data[0].assessment_marks[parseInt(k, 10) + 1]) {
-            this.individualDataSet.data.push(parseFloat(res.data[0].assessment_marks[parseInt(k, 10) + 1].percentage));
-          } else {
-            this.individualDataSet.data.push(null);
-          }
-        });
-        chartInfo['datasets'] = [this.individualDataSet, this.averageDataSet];
-        if (this.medianDataSet.data.some(m => m !== null)) {
-          chartInfo['datasets'].push(this.medianDataSet);
-        }
-      } else {
-        Object.keys(res.schemes).forEach(k => {
-          this.averageDataSet.data.push(res.schemes[k].average ? res.schemes[k].average : null);
-          this.medianDataSet.data.push(res.schemes[k].median ? res.schemes[k].median : null);
-        });
-        chartInfo['datasets'] = [this.averageDataSet, this.medianDataSet];
-      }
+    }).then(res => this.setState({loading: false, ...res}))
+  }
 
-      res.columns.forEach((c) => {
-        if (res.assessment_info[c.Header]) {
-          c.Header += ` (/${Math.round(parseFloat(res.assessment_info[c.Header]['total']) * 100) / 100})`;
-        }
-      });
-      this.setState({
-        data: res.data,
-        columns: res.columns,
-        loading: false,
-        ...chartInfo
-      });
-    });
+  compileDatasets = () => {
+    if (this.props.student) {
+      return [
+        {...this.averageDataSet, data: this.state.average_data},
+        {...this.individualDataSet, data: this.state.individual_data}
```

Comment:
The median is allowed to be shown to students for assignments that have the "display median to students" attribute set to true. (You should also check that the median data for students should have `null` for the other assessments, but the correct stats for any assignments where this attribute is true.)